### ISSUE TITLE
 Detect Player Given Commands in Script

### ISSUE THREAD

**Issue by arosenow** [gemrb/gemrb#2253]
#### Description
Allow scripting to detect if you have given a command to the player with the script applied.
Allow scripts to change local variables without overriding the current action of the player

#### Additional context
When making scripts for archers there is currently one bugbear, if you make an archer run to help someone you can only do so via a move to that character, that means if they see an enemy on the way you must directly override any current action (eg player commanded movement) to make them attack. If the scripts could detect if you'd given them an order and then change a local variable it could then turn off attacking via script until they have an empty action list

**Comment by lynxlynxlynx** [comment]
Did you try something like this condition:
```
!ActionListEmpty()
See([ENEMY])
```
... and then attack the nearest.


On the question of "allow scripts to change ... without overriding the current action": that's impossible, since the only way scripts can change anything is by running an action (if you ignore the triggers with side effects).

**Comment by arosenow** [comment]
It is very easy to make a character attack whilst their action list isn't empty, however if you then give the character a command, most players would rather the script obeyed the latest command. 

The problem in existing IE is the script itself must be permitted to override any commands you gave it temporarily, because it is moving towards their companion (action list already isn't empty), so it will override any new order you give them to attack instead if you've previously told them to start moving to a call for help, unless you can disable the special exception you've granted in a way that also doesn't override the latest command (toggling local vars currently also is an action so it overrides your command too)

**Comment by arosenow** [comment]
Whether it would be possible to distinguish if the current action is player initiated may resolve the issue. Eg ActionListPlayerInitEmpty()

**Comment by lynxlynxlynx** [comment]
Yeah, you're right, a negated ActionListEmpty can never be satisfactory.
All I can come up with is having the action generated by the user click not be Attack, but AttackReevaluate, if a global is set (and it would be the period parameter). But that would also not preserve the original target.
